utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java

```diff
@@ -322,4 +328,15 @@ public static String appendUri(String baseUri, String path) {
     public static boolean isSingleHeader(String h) {
         return SINGLE_HEADERS.contains(StringUtils.lowerCase(h));
     }
+
+    /**
+     * Extracts query parameters from the given URI
+     */
+    public static Map<String, List<String>> uriParams(URI uri) {
+        return Pattern.compile(QUERY_PARAM_PATTERN)
```

Comment:
BTW, I think a more accurate name would be "QUERY_PARAM_DELIMETER_PATTERN"

Reply:
the split is safe to use in my opinion since it uses a StringTokenizer if only a single character is provided since java 7 according to http://hg.openjdk.java.net/jdk7/jdk7/jdk/rev/1ff977b938e5